### ISSUE TITLE
How would this work with bootstrap themes like bootswatch

### ISSUE THREAD

**Issue by mattapperson** [FormidableLabs/radium-bootstrap#15]
I think this project (and obviously radium) are REALLY great... but how do you deal with things like bootswatch themes when using this and radium?

**Comment by alexlande** [comment]
@rgerstenberger has done some preliminary work in this direction which is very cool: https://github.com/FormidableLabs/radium-bootstrap/blob/master/modules/util/configure-settings.js

We're definitely interested in supporting existing Bootstrap themes (and providing more customization) in the future.

**Comment by mattapperson** [comment]
Yea, the issue with this (as much as I truly do love it) is that it results in inline styles. So themes would have to be converted (a pain I am sure). Perhaps a better solution would be to use react to generate a style tag in the dom with dynamically created css rather then inlining? Then radium would just create dynamically an id or class used by the dynamic css

Just a thought...

**Comment by alexlande** [comment]
Yeah, there are definitely benefits to the dynamic `style` element idea. It's something we're exploring in @FormidableLabs/radium for certain behavior, but using it for everything isn't on the roadmap at this point.

As far as Bootstrap goes, the plan for now is to support Bootstrap variables as JSON, which should be helpful for starting to port over themes at least. From there my hope is that we can find a good interface for adding app-specific styles directly to a given type of component.

On a related note, have you checked out https://github.com/js-next/react-style? It might be closer to what you're looking for. You could also potentially use Radium *with* react-style, maybe. I haven't actually tried that but it seems viable.

**Comment by colinmegill** [comment]
+1 to this - @mattapperson this is obviously a non-trivial amount of work. 

I had the same gut reaction as Alex - seems like react-style might be what you're looking for, as the benefits to Radium, and this cannot be emphasized enough, are *dynamic computation*. Not every site needs this, but the webapps we build desperately need a better solution than CSS. 

The benefits to rewriting are substantial - imagine being able to make an async call for a theme to an API endpoint that returns JSON, and in the callback restyling your entire app. Or postmessaging an object into an `<iframe>` widget and styling it that way. Those are just a couple, but there are some pretty powerful adds to _what themes could do in general_ if you add computation. For some people, I expect it will be worth the rewrite.
